locator/tablets.cc

```diff
@@ -207,7 +208,14 @@ std::optional<tablet_info> merge_tablet_info(tablet_info a, tablet_info b) {
     }
 
     auto repair_time = std::max(a.repair_time, b.repair_time);
-    return tablet_info(std::move(a.replicas), repair_time, a.repair_task_info, a.migration_task_info);
+
+    int64_t sstables_repaired_at = initial_sstables_repaired_at_after_merge;
+    auto info = tablet_info(std::move(a.replicas), repair_time, a.repair_task_info, a.migration_task_info, sstables_repaired_at);
+    // Record the sstables_repaired_at before merge, so we can use it later to
+    // rewrite sstable.repaired_at for the merged tablets.
+    info.sstables_repaired_at_before_merge.emplace(left, a.sstables_repaired_at);
+    info.sstables_repaired_at_before_merge.emplace(right, b.sstables_repaired_at);
+    return info;
```

Comment:
Why do we need this? Isn't it enough for compaction to propagate the minimum across input sstables? It can happen tablet 0 before merge was repaired at 10, tablet 1 before merge repaired at 5. so sstables from 0 and 1 compacted together should yield an output with repaired at 5. It seems we can handle this in replica layer. Alternatively, we have the sstable storing a map of token_range -> repaired_at. Then repair would read from the sstable only the range which repaired_at doesn't satisfy the condition. We'd merge ranges in the map when they have the same value, to prevent it from exploding over time. Maybe it's enough to just propate the minimum across input.

Reply:
Ok I learned from Botodn that repaired_at just makes sense within its own namespace, and when merging, we cannot just pick one value from tablet 0 and make it work on sstables produced by tablet 1 (before merge)

Reply:
I thought using std::max() should be sufficient.